src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskReplicationMIControllerTest.java

```diff
@@ -228,29 +394,69 @@ public void user_should_claim_task_and_claim_action_recorded_in_replicate_db() {
+        assertTrue(OffsetDateTime.parse(claimJsonPathEvaluator.get("reportable_task_list.get(0).updated"))
+                       .isBefore(OffsetDateTime.parse(reClaimJsonPathEvaluator.get("reportable_task_list.get(0).updated"))));
+        assertEquals(claimJsonPathEvaluator.get("reportable_task_list.get(0).created").toString(),
+                     reClaimJsonPathEvaluator.get("reportable_task_list.get(0).created").toString());
+        assertEquals(claimJsonPathEvaluator.get("reportable_task_list.get(0).wait_time_days").toString(),
+                     reClaimJsonPathEvaluator.get("reportable_task_list.get(0).wait_time_days").toString());
+        assertEquals(claimJsonPathEvaluator.get("reportable_task_list.get(0).wait_time").toString(),
+                     reClaimJsonPathEvaluator.get("reportable_task_list.get(0).wait_time").toString());
+        assertEquals(claimJsonPathEvaluator.get("reportable_task_list.get(0).first_assigned_date_time").toString(),
+                     reClaimJsonPathEvaluator.get("reportable_task_list.get(0).first_assigned_date_time").toString());
+        assertEquals(claimJsonPathEvaluator.get("reportable_task_list.get(0).first_assigned_date").toString(),
+                     reClaimJsonPathEvaluator.get("reportable_task_list.get(0).first_assigned_date").toString());
+
         resultReport.prettyPrint();
         resultReport.then().assertThat()
             .statusCode(HttpStatus.OK.value())
             .body("reportable_task_list.size()", equalTo(1))
             .body("reportable_task_list.get(0).state", equalTo("ASSIGNED"))
-            .body("reportable_task_list.get(0).number_of_reassignments", equalTo(1));
-
-        resultAssignmentsClaim.then().assertThat()
-            .statusCode(HttpStatus.OK.value())
-            .body("task_assignments_list.size()", equalTo(2))
-            .body("task_assignments_list.get(0).assignment_end_reason", equalTo("UNCLAIMED"))
-            .body("task_assignments_list.get(1).assignment_end_reason", equalTo(null));
+            .body("reportable_task_list.get(0).number_of_reassignments", equalTo(1))
+            .body("reportable_task_list.get(0).assignee", notNullValue())
+            .body("reportable_task_list.get(0).updated_by", notNullValue())
+            .body("reportable_task_list.get(0).updated", notNullValue())
+            .body("reportable_task_list.get(0).update_action", equalTo("Claim"))
+            .body("reportable_task_list.get(0).created_date", notNullValue())
+            .body("reportable_task_list.get(0).due_date", notNullValue())
+            .body("reportable_task_list.get(0).last_updated_date", notNullValue())
+            .body("reportable_task_list.get(0).first_assigned_date", notNullValue())
+            .body("reportable_task_list.get(0).first_assigned_date_time", notNullValue())
+            .body("reportable_task_list.get(0).wait_time_days", equalTo(0))
+            .body("reportable_task_list.get(0).wait_time", notNullValue());
```

Comment:
Maybe it would be better to test whether dates are null before you have the specific dates checks above to avoid NPEs?
Should do the same in the other tests below.

Reply:
Done.